experimental/SetPartitions/src/PartitionProperties.jl

```diff
@@ -0,0 +1,190 @@
+            block_to_size[i] = 1
+        else
+            block_to_size[i] = get(block_to_size, i, -1) + 1
+            if get(block_to_size, i, -1) > 2
+                return false
+            end
+        end
+    end
+    return all(i -> i == 2, values(block_to_size))
+end
+
+"""
+    is_balanced(p::T) where {T<:Union{SetPartition, ColoredPartition}}
+
+Return whether `p` is a balanced partition.
+
+# Examples
+```jldoctest
+julia> is_balanced(SetPartition([1, 2, 3], [3, 2, 1]))
+true
+```
+"""
+function is_balanced(p::T) where {T<:Union{SetPartition, ColoredPartition}}
+
+    p_vector = vcat(upper_points(p), lower_points(p))
+
+    # remember length lower points
+    upper = length(upper_points(p))
+    upper_uneven = upper % 2 == 1 ? true : false
+
+    # Dictionary from block to sum of -1 (repr odd indices) and 1 (repr even indices)
+    block_to_size = Dict{Int, Int}()
+
+    # prefill dict with zeros
+    for i in p_vector
+        block_to_size[i] = 0
+    end
+
+    # Initialize dictionary
+    for (i, n) in enumerate(p_vector)
+        condition = upper_uneven ? i % 2 == 1 : (i <= upper ? i % 2 == 1 : i % 2 == 0)
+        if condition
+            block_to_size[n] = get(block_to_size, n, -1) - 1
+        else
+            block_to_size[n] = get(block_to_size, n, -1) + 1
+        end
+    end
+
+    return all(i -> i == 0, values(block_to_size))
+end
+
+"""
+    is_noncrossing(p::T) where {T<:Union{SetPartition, ColoredPartition}}
+
+Return whether `p` is a non-crossing partition.
+
+# Examples
+```jldoctest
+julia> is_noncrossing(SetPartition([1, 2, 2, 3, 1, 4], [4, 3]))
+false
+```
+"""
+function is_noncrossing(p::T) where {T<:Union{SetPartition, ColoredPartition}}
+
+    # transform partition to only upper points
+    p_vector = vcat(upper_points(p), reverse(lower_points(p)))
+
+    # Dictionary from block to size of block
+    block_to_size = Dict{Int, Int}()
+
+    # Initialize dictionary
+    for i in p_vector
+        if !(i in keys(block_to_size))
+            block_to_size[i] = 1
+        else
+            block_to_size[i] = get(block_to_size, i, -1) + 1
+        end
+    end
+
+    # blocks we have already seen in the iteration process
+    already_seen = Set{Int}()
+    last_incompleted = []
+    incompleted = Set{Int}()
+
+    for (i, n) in enumerate(p_vector)
+        if (n in incompleted && 
+            (isempty(last_incompleted) ? -1 : last_incompleted[end]) != n)
+            return false
+        end
+        if !(n in already_seen)
+            push!(already_seen, n)
+            block_to_size[n] = get(block_to_size, n, -1) - 1
+            if (get(block_to_size, n, -1) > 0 && 
+                (isempty(last_incompleted) ? -1 : last_incompleted[end]) != n)
+                push!(last_incompleted, n)
+                push!(incompleted, n)
+            end
+        else
+            if p_vector[i-1] != n && get(block_to_size, p_vector[i-1], -1) != 0
+                return false
+            else
```

Comment:
Since you `return` in the line above, you can just `end` here, and then de-indent the following block. Advantage: the code is nested less, which makes it either to read and understand.
```suggestion
            end
```